shared/api/models/DateModel.ts

```diff
@@ -128,23 +138,16 @@ class DateModel {
     return null
   }
 
-  private rruleToDateTime(date: Date): DateTime {
-    const dateTime = DateTime.fromJSDate(date)
-    // rrule is not correctly handling timezones and always returning local time in the shape of UTC
-    // Therefore we have to manually apply timezone offset, e.g. for daylight savings time
-    // https://github.com/jkbrzt/rrule#important-use-utc-dates
-    return dateTime.plus({ minutes: this._offset - dateTime.offset })
-  }
-
-  private currentDateToRrule(): Date {
-    // rrule is not correctly handling timezones and is expecting dates in "local time", i.e. in the timezone of dstart
-    // Therefore we have to manually apply timezone offset, e.g. for daylight savings time
-    // https://github.com/jkbrzt/rrule#important-use-utc-dates
-    // Also include dates happening at the moment by subtracting the duration
-    return DateTime.now()
-      .minus(this._duration)
-      .minus({ minutes: this._offset - DateTime.now().offset })
-      .toJSDate()
+  private getRecurrenceRuleInLocalTime(recurrenceRule: RRuleType): RRuleType | undefined {
```

Comment:
:wrench: No need for `undefined`. Please also remove check above.
```suggestion
  private getRecurrenceRuleInLocalTime(recurrenceRule: RRuleType): RRuleType {
```

Reply:
Good catch, thanks! Too much refactoring 🤪